### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Mohitk391**
Authentication and History Components were added.

Files to check:
1) Login.js
2) Signup.js
3) History.js
4) HistoryContext.js
5) UserContext.js
6) HistoryReducer.js
7) UserReducer.js

preview:  https://62d711e5ef5eac115a96642e--lucky-sopapillas-2b808e.netlify.app/

### CODE REVIEW

File: src/contexts/PlaylistContext.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                    }
+                );
+                playlistDispatch("SET_PLAYLIST", response.data.playlists);
+            }
+            catch(error){
+                console.error(error);
+            }
+        };
+        fetchPlaylist();
+    },[token]);
```
Comment (MayankKumar10) [inline src/contexts/PlaylistContext.js:30]: Look into indentation here

File: src/pages/playlist/playlist.css [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    height: 100vh;
+    color: var(--white);
+    /* visibility: hidden;this class should be replaced by overlay layer class from component library, once it is updated there.  */
+}
+
+.create-playlist{
+    background-color: var(--white);
+    color: var(--black);
+    padding: 1rem;
+    border-radius: 10px;
```
Comment (MayankKumar10) [inline src/pages/playlist/playlist.css:60]: use rem instead of px

**Review by MayankKumar10 — APPROVED** [review]
Look into these Suggestions everything else looks good to me you can merge it.

**Comment by varun-Tad** [comment]
Looks good.You can go ahead and merge

**Pull request merged**